### DISCUSSION TITLE
429 errors handling

### DISCUSSION THREAD

**Discussion by yw791772** [langchain-ai/langserve#473] [Q&A] [answered]
Hi, I'm following the [astream method](https://github.com/langchain-ai/langserve/blob/54725a91edece67a61ac880f9f02239b6d81dd42/langserve/client.py#L617-L668) in the langserve client code, and I have a try catch around the async sse code.
For testing purpose, I use an api key that has no quota, and I can see the 429 errors from the langserve app logs.
But when I looking at the decoded message from [it](https://github.com/langchain-ai/langserve/blob/54725a91edece67a61ac880f9f02239b6d81dd42/langserve/client.py#L653) on the client side, I only see 500 with internal server error.

Any guidance on how I can surface the 429 errors on the client side, so I can build specific around this error

**Comment by eyurtsev** [comment, marked as answer]
Opened an issue for this:  https://github.com/langchain-ai/langserve/issues/481

Fixing this behavior will also help likely: https://github.com/langchain-ai/langserve/issues/414

For now, feel free to fork the code and updating the lines here: https://github.com/langchain-ai/langserve/blob/e1cec832da29173df89bfe50b35be8ce59b1a8b5/langserve/api_handler.py#L1214-L1214
